src/DiscordBot.py

```diff
@@ -71,21 +71,25 @@ async def health_check_task():
 
 
 @tasks.loop(seconds=60)
-async def scheduled_task(
-    internship_github: GitHubUtilities, newgrad_github: GitHubUtilities, job_utilities: JobsUtilities
-):
+async def scheduled_task(job_utilities: JobsUtilities):
     """
     A scheduled task that runs every 60 seconds to check for new commits in the GitHub repository.
 
     Parameters:
-        - internship_github: Internship object
-        - newgrad_github: New Grad GitHubUtilities object
         - job_utilities: The JobsUtilities object
     """
     async with lock:
         try:
             start_time = datetime.now()
 
+            latest_repo = JobsUtilities.get_latest_internship_repo()
```

Comment:
Still concerned about this line here. Why do we need to make an api call every minute to check the latest repo? Why not first try to make a connection to latest cached repo and if it doesn't exist, loop until we can find the latest repo. Thoughts?

Reply:
I think that's reasonable. I'll see what I can do

Reply:
Could you explain further?

Reply:
I agree that it is reasonable to try to save the repo somewhere and only get the latest repo whenever the one we have saved is outdated. In my most recent commit (c8483df2ad57bf786eb930da467aca153ccacb88) I save the repo and always try to use what I have saved in the variable. If the repo we have saved no longer works than I fetch the latest repo again